clio-client/src/main/scala/org/broadinstitute/clio/client/webclient/ClioWebClient.scala

```diff
@@ -88,4 +90,21 @@ class ClioWebClient(clioHost: String, clioPort: Int, useHttps: Boolean)(
       ).addHeader(Authorization(credentials = OAuth2BearerToken(bearerToken)))
     )
   }
+
+  def getResponseAsJson(httpResponse: Future[HttpResponse]): Future[Json] = {
+    httpResponse.flatMap(Unmarshal(_).to[Json])
+  }
+
+  def ensureOkResponse(
+    httpResponse: Future[HttpResponse]
+  ): Future[HttpResponse] = {
+    httpResponse.flatMap(
```

Comment:
You could use `map` instead of `flatMap` here, since you're always mapping success to `Future.success` and failure to `Future.failed`. If you `throw` inside a `map` it'll result in a failed `Future`.

Reply:
good to know!